test/e2e/common/node/pod_level_resources.go

```
@@ -0,0 +1,319 @@
+				qos:          v1.PodQOSBurstable,
+				podResources: &ResourceInfo{CPUReq: "50m", CPULim: "100m", MemReq: "50Mi", MemLim: "100Mi"},
+			},
+		},
+		{
+			name:         "Burstable QoS pod, 1 container with resources",
+			podResources: &ResourceInfo{CPUReq: "50m", CPULim: "100m", MemReq: "50Mi", MemLim: "100Mi"},
+			containers: []ContainerInfo{
+				{Name: "c1", Resources: &ResourceInfo{CPUReq: "20m", CPULim: "100m", MemReq: "50Mi", MemLim: "100Mi"}},
+				{Name: "c2"},
+			},
+			expected: expectedPodConfig{
+				qos:          v1.PodQOSBurstable,
+				podResources: &ResourceInfo{CPUReq: "50m", CPULim: "100m", MemReq: "50Mi", MemLim: "100Mi"},
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		ginkgo.It(tc.name, func(ctx context.Context) {
+			podClient := e2epod.NewPodClient(f)
+			tStamp := strconv.Itoa(time.Now().Nanosecond())
+			podMetadata := &metav1.ObjectMeta{Name: "testpod", Namespace: f.Namespace.Name, Labels: map[string]string{"time": tStamp}}
+			testPod := makePod(podMetadata, tc.podResources, tc.containers)
+
+			ginkgo.By("creating pods")
+			pod := podClient.CreateSync(ctx, testPod)
+
+			ginkgo.By("verifying pod resources are as expected")
+			VerifyPodResources(*pod, *tc.podResources)
+
+			ginkgo.By("verifying pod QoS as expected")
+			VerifyQoS(*pod, tc.expected.qos)
+
+			ginkgo.By("verifying pod cgroup values")
+			err := VerifyPodCgroups(ctx, f, pod, *tc.podResources)
+			framework.ExpectNoError(err, "failed to verify pod's cgroup values: %v", err)
+
+			ginkgo.By("verifying containers cgroup limits are same as pod container's cgroup limits")
+			err = VerifyContainersCgroupLimits(f, pod)
+			framework.ExpectNoError(err, "failed to verify containers cgroup values: %v", err)
+
+			ginkgo.By("deleting pods")
+			delErr := e2epod.DeletePodWithWait(ctx, f.ClientSet, pod)
+			framework.ExpectNoError(delErr, "failed to delete pod %s", delErr)
+		})
+	}
+}
+
+func VerifyContainersCgroupLimits(f *framework.Framework, pod *v1.Pod) error {
```

nit: instead of returning error on the first failure, you might want to aggregate the errors (or return a list of errors) to facilitate debugging.